mobile/lib/common/dlc_channel_change_notifier.dart

```diff
@@ -177,7 +177,18 @@ class DlcChannelChangeNotifier extends ChangeNotifier implements Subscriber {
   void notify(bridge.Event event) {
     if (event is bridge.Event_DlcChannelEvent) {
       DlcChannel channel = DlcChannel.fromApi(event.field0);
-      channels[channel.id] = channel;
+
+      if (channels.containsKey(event.field0.referenceId)) {
+        // if we can find the channel on the reference id we can remove it now as we have the correct
+        // channel id
+        channels.remove(event.field0.referenceId);
+      } else if (event.field0.channelState is bridge.ChannelState_Offered) {
+        // if the channel state is in offered we save the channel on the reference id as the channel
+        // id will change.
+        channels[event.field0.referenceId] = channel;
+      } else {
+        channels[channel.id] = channel;
+      }
```

Comment:
I don't think that's a good idea. This is merely for the frontend. If we'd move that logic to the backend we'd need to keep a state about what the frontend already knows and what not. I'd rather not do that.

Note, this is still an issue because we do not have a long living id that will identify a dlc channel uniquely. The reference id helps to link the temporary channel id and the actually channel id, but it will always require logic like this to get a unique identifier of a dlc channel

Reply:
I guess that would be solved with a dedicated `UserChannelId`.